src/components/YourCompany/OfficeAddresses.vue

```diff
@@ -624,7 +629,7 @@ export default class OfficeAddresses extends Mixins(CommonMixin) {
    * sets local properties and emits state events.
    */
   @Watch('getOfficeAddresses', { deep: true })
```

Comment:
Why not use `immediate: true`  to fire this method up front?

Reply:
It's fired up front already. The issue is during the re-mount ie changing to review and back to company changes. 
Would immediate fire it on a re-mount?

Reply:
There is no such thing as a remount. The component is actually being re-loaded/re-rendered.

(This is part of why NR was nasty -- the components are always (mostly) rendered, and conditionally displayed. In that case you have to catch the display event and init there.)

Reply:
Short answer: yes, when the component is reloaded, all immediate watchers should fire.

What is firing this up front already? Why doesn't that do what it's supposed to with the data?

Reply:
> There is no such thing as a remount. The component is actually being re-loaded/re-rendered.
> 
> (This is part of why NR was nasty -- the components are always (mostly) rendered, and conditionally displayed. In that case you have to catch the display event and init there.)

I'm not sure what you want to call it, re-mounting seemed most accuratre. The component itself is mounted every time you change to Review and back to the components.

It fires up front because the store values change with regards to the office addresses, every time a filing is loaded. Why it works initially but doesn't when it 'mounts again'. Because there is no changes to the store value at that point.

Reply:
> I'm not sure what you want to call it, re-mounting seemed most accuratre. The component itself is mounted every time you change to Review and back to the components.

Haha, I agree with you there. Let's call it "remounted" then.

My point is that this component is destroyed between mounted calls. It's not just dormant and re-initialized.

Reply:
What if this component wasn't rendered (mounted!) until the data is loaded...?

Reply:
Totally, the issue was purely that the initialization was based on a watcher (office address store data) which only changes once in an Alteration. Not an issue in corrections because the single view, the component is never destroyed.

I think why this isn't working like our other components, is the value fed into the Base Address Components isn't exactly reactive, it is computed values which the component is internally doing, and in this case the trigger for that is a change.

I'll try `immediate` though. Would much prefer that as well!

Reply:
immediate works great, thanks! 

I didn't know this option did this, assumed it tied into `created` hook, this is really good to know 👍